### ISSUE TITLE
Better implementation of download process

### ISSUE THREAD

**Issue by bernardodsanderson** [bernardodsanderson/vido#18]
Need a better implementation of the download process so the user can see the progress of the download.

**Comment by ninpnin** [comment]
The OS also gives me a "program not responding" prompt while downloading.

**Comment by ryonakano** [comment]
@ninpnin Yeah it seems currently the download process is done in a "sync" process so it appears to be freezes, while it's actually working in the background.
